### PR TITLE
Upgrade `undici` dependency (`5.1.1` → `5.5.1`).

### PR DESCRIPTION
**Pull request by azasypkin**
## Summary

Upgrade `undici` dependency (`5.1.1` → `5.5.1`).

Change log: https://github.com/nodejs/undici/releases
Change set: https://github.com/nodejs/undici/compare/v5.1.1...v5.5.1

__Note to reviewers:__ It doesn't seem there were any breaking changes since `v5.1.1`, `npm test` doesn't complain either. We (Kibana) are particularly interested in upgrade to `v5.1.1` even though we're not _yet_ affected by the issue this release had fixed (we don't seem to use `ProxyAgent` from `undici` anywhere).

### CODE REVIEW

File: src/Transport.ts [hunk 1]
```diff
@@ -473,7 +473,7 @@ export default class Transport {
         result.headers = headers
 
         if (this[kProductCheck] != null && headers['x-elastic-product'] !== this[kProductCheck] && statusCode >= 200 && statusCode < 300) {
-          throw new ProductNotSupportedError(this[kProductCheck] as string, result)
```
Comment (azasypkin) [inline src/Transport.ts:476]: __note:__ `npm test` (specifically `ts-standard`) was complaining about unnecessary assertion.

File: package.json [hunk 2]
```diff
@@ -62,7 +62,7 @@
     "ms": "^2.1.3",
     "secure-json-parse": "^2.4.0",
     "tslib": "^2.4.0",
-    "undici": "^5.1.1"
+    "undici": "^5.5.1"
```
Comment (azasypkin) [inline package.json:65]: __note:__ let me know if I need to bump version of the `@elastic/transport` package itself as well.

**Comment by azasypkin** [comment]
Just realized that @delvedor is no longer on the Clients team, @sethmlarson would you mind suggesting someone on your team to review this PR?

Thanks!

**Review by delvedor — APPROVED** [review]
LGTM

I would also note that this update shouldn't be necessary.
npm always installs the latest minor.patch version of a given major for dependencies.

If you run `npm install @elastic/transport` in an empty folder, you should see the latest version of undici inside `node_modules`.

**Comment by azasypkin** [comment]
> I would also note that this update shouldn't be necessary.
npm always installs the latest minor.patch version of a given major for dependencies.
> If you run npm install @elastic/transport in an empty folder, you should see the latest version of undici inside node_modules.

Correct, I'm bumping the version in the `package.json` just for the sake of order and to make sure the old version isn't stuck in the consumers' `.lock` files.

**Pull request merged**